src/data_api/orders.se

```diff
@@ -155,105 +155,79 @@ def getWorstBidOrderID(market: address, outcome: uint256):
-            if(betterOrderID == 0):
-                ~invalid()
-            if(fxpPrice > self.orders[betterOrderID].fxpPrice):
-                ~invalid()
-    return(1: uint256)
+# @public
+def isBetterPrice(type: uint256, fxpPrice, orderID: address):
+    if(type == BID):
+        return((fxpPrice > self.orders[orderID].fxpPrice): uint256)
+    return((fxpPrice < self.orders[orderID].fxpPrice): uint256)
 
-# @private
-def updateWorstBidOrder(orderID: address, market: address, fxpPrice, outcome: uint256, worseOrderID: uint256):
-    if(msg.sender != self):
-        ~invalid()
-    worstBidOrderID = self.worstOrder[market][outcome].bid
-    # If worst bid is not set, this order is the new worst bid
-    if(worstBidOrderID == 0):
-        self.worstOrder[market][outcome].bid = orderID
-    else:
-        # If price lower than worst bid price, this order is the new worst bid
-        if(fxpPrice < self.orders[worstBidOrderID].fxpPrice):
-            if(worseOrderID != 0):
-                ~invalid()
-            self.worstOrder[market][outcome].bid = orderID
-        # If price is higher than the worst bid price:
-        #  - worseOrderID must be specified
-        #  - price must be greater than or equal to worseOrderID's price
-        else:
-            if(worseOrderID == 0):
-                ~invalid()
-            if(fxpPrice < self.orders[worseOrderID].fxpPrice):
-                ~invalid()
-    return(1: uint256)
+# @public
+def isWorsePrice(type: uint256, fxpPrice, orderID: address):
+    if(type == BID):
+        return((fxpPrice < self.orders[orderID].fxpPrice): uint256)
+    return((fxpPrice > self.orders[orderID].fxpPrice): uint256)
 
-# @private
-def updateBestAskOrder(orderID: address, market: address, fxpPrice, outcome: uint256, betterOrderID: uint256):
-    if(msg.sender != self):
-        ~invalid()
-    bestAskOrderID = self.bestOrder[market][outcome].ask
-    # If best ask is not set, this order is the new best ask
-    if(bestAskOrderID == 0):
-        self.bestOrder[market][outcome].ask = orderID
+# @public
+def ascendOrderList(type: uint256, fxpPrice, lowestOrderID: address):
+    worseOrderID = lowestOrderID
+    if(type == BID):
+        isWorstPrice = fxpPrice <= self.orders[worseOrderID].fxpPrice
     else:
-        # If price lower than best ask price, this order is the new best ask
-        if(fxpPrice < self.orders[bestAskOrderID].fxpPrice):
-            if(betterOrderID != 0):
-                ~invalid()
-            self.bestOrder[market][outcome].ask = orderID
-        # If price is higher than the best ask price:
-        #  - betterOrderID must be specified
-        #  - price must be greater than than or equal to betterOrderID's price
-        else:
-            if(betterOrderID == 0):
-                ~invalid()
-            if(fxpPrice < self.orders[betterOrderID].fxpPrice):
-                ~invalid()
-    return(1: uint256)
+        isWorstPrice = fxpPrice >= self.orders[worseOrderID].fxpPrice
```

Comment:
The difference is intentional in this case, since this function walks _up_ the order list.  i.e., if a new order has an equivalent price to an existing order, it's considered "worse" so that we respect price-time priority.